src/main/java/org/takes/rq/RqForm.java

```diff
@@ -233,4 +232,48 @@ public String single(final CharSequence name, final String def) {
             return value;
         }
     }
+
+    /**
+     * Fake RqForm accepting parameters in the constructor.
+     * @author Matteo Barbieri (barbieri.matteo@gmail.com)
+     */
+    @EqualsAndHashCode(callSuper = true, of = "map")
+    final class Fake extends RqWrap implements RqForm {
+
+        /**
+         * Map of params and values.
+         */
+        private final transient ConcurrentMap<String, List<String>> map;
+
+        /**
+         * Ctor.
+         * @param req Original request
+         * @param params Parameters list
+         */
+        public Fake(final Request req, final String... params) {
+            super(req);
+            if (params.length % 2 != 0) {
+                throw new IllegalArgumentException(
+                    "Wrong number of parameters"
+                );
+            }
+            this.map = new ConcurrentHashMap<String, List<String>>(0);
```

Comment:
@yegor256 I don't think converting from an array to a string and then parse the string to create a map is good. I would probably create another one accepting a map (the most obvious data structure for parameters) and then refactor `RqWithBody` and `Fake` to use that.